lib/main.js

```diff
@@ -63,6 +63,18 @@ export default {
         this.useLocalTslint = use;
       }),
     );
+
+    this.subscriptions.add(
+      atom.workspace.observeTextEditors((editor) => {
+        editor.onDidSave(() => {
+          if (atom.config.get('linter-tslint.fixOnSave')) {
```

Comment:
Your issue is that you need to check the grammar scope on the saving editor, not only `fixOnSave`